Cargo.toml

```diff
@@ -70,7 +70,10 @@ license = "Apache-2.0"
 clippy.print_stdout = "warn"
 clippy.print_stderr = "warn"
 clippy.implicit_clone = "warn"
+clippy.readonly_write_lock = "allow"
```

Comment:
This should be a todo that we later revisit if locks can improved. And we should suppress the warning in place where it's not the case.

We can create an issue for it.